### DISCUSSION TITLE
Dahua camera

### DISCUSSION THREAD

**Discussion by esalasamaya** [awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#1024] [Q&A]
I have a dahua camera, the rtsp ends with "/cam/realmonitor?channel=1&subtype=1" and does not work.

LOG:
New pad found: recv_rtp_src_0_2832907471_96
[INFO ] [08-06-2023 17:00:25:828.586 GMT] Pad link successful
[INFO ] [08-06-2023 17:00:25:849.136 GMT] kinesisVideoStreamFormatChanged(): Stream format changed.
[INFO ] [08-06-2023 17:00:25:879.776 GMT] putStreamResultEvent(): Put stream result event. New upload handle 0
[DEBUG] [08-06-2023 17:00:26:576.567 GMT] postReadCallback(): Pausing CURL read for upload handle: 0
[INFO ] [08-06-2023 17:00:26:576.708 GMT] writeHeaderCallback(): RequestId: dc4eeaff-97a9-87hh-80b2-e1965847159b
[WARN ] [08-06-2023 17:00:55:917.097 GMT] curlCompleteSync(): curl perform failed for url https://s-444441da.kinesisvideo.us-east-1.amazonaws.com/putMedia with result Timeout was reached: Operation too slow. Less than 30 bytes/sec transferred the last 30 seconds
[WARN ] [08-06-2023 17:00:55:917.405 GMT] curlCompleteSync(): HTTP Error 0 : Response: (null)
Request URL: https://s-468461da.kinesisvideo.us-east-1.amazonaws.com/putMedia
Request Headers:
    Authorization: AWS4-HMAC-SHA256 Credential=ASIA000093SWE2NXVS/20230608/us-east-1/kinesisvideo/aws4_request, SignedHeaders=connection;host;transfer-encoding;user-agent;x-amz-date;x-amzn-fragment-acknowledgment-required;x-amzn-fragment-timecode-type;x-amzn-producer-start-timestamp;x-amzn-stream-name, Signature=baf13d666uuuuuuuucf650eb8749b16d2e4c1d617
[DEBUG] [08-06-2023 17:00:55:917.443 GMT] putStreamCurlHandler(): Network thread for Kinesis Video stream: CAMERA-01 with upload handle: 0 exited. http status: 0
[WARN ] [08-06-2023 17:00:55:917.558 GMT] putStreamCurlHandler(): Stream with streamHandle 94171141647856 uploadHandle 0 has exited without triggering end-of-stream. Service call result: 599
[INFO ] [08-06-2023 17:00:55:917.593 GMT] kinesisVideoStreamTerminated(): Stream 0x55a5ed722df0 terminated upload handle 0 with service call result 599.
[DEBUG] [08-06-2023 17:00:55:917.617 GMT] defaultStreamStateTransitionHook(): Stream state machine retry count: 0
[DEBUG] [08-06-2023 17:00:55:917.636 GMT] defaultStreamStateTransitionHook(): 
 KinesisVideoStream base result is [599]. Executing KVS retry handler of retry strategy type [1]
[DEBUG] [08-06-2023 17:00:55:972.137 GMT] defaultStreamStateTransitionHook(): Stream state machine retry count: 1
[DEBUG] [08-06-2023 17:00:55:972.299 GMT] defaultStreamStateTransitionHook(): 
 KinesisVideoStream base result is [599]. Executing KVS retry handler of retry strategy type [1]
[DEBUG] [08-06-2023 17:00:55:972.442 GMT] streamReadyHandler invoked

NOTE: I tried with another camara different to Dahua and all works correctly, so the configuration is correct. To resolve the special character "&", the command is  ./kvs_gstreamer_sample CAMERA-NAME "rtsp://user:pass@IP:PORT/cam/realmonitor?channel=1&subtype=1".Would you know what is the error?

**Comment by disa6302** [comment]
@esalasamaya , if a camera other than the non-working worked, then the issue is probably with the frames coming out of this specific URL. Would suggest trying out this camera with a filesink/fakesink/autovideosink to ensure the RTSP URL works in a gstreamer pipeline before trying out the sample. 
Also ensure the region of the camera and the region where this sample is being run is the same.
